receiver/k8sobjectsreceiver/receiver.go

```diff
@@ -146,40 +147,85 @@ func (kr *k8sobjectsreceiver) startWatch(ctx context.Context, config *K8sObjects
 	kr.stopperChanList = append(kr.stopperChanList, stopperChan)
 	kr.mu.Unlock()
 
+	resourceVersion := config.ResourceVersion
+	var err error
+	if resourceVersion == "" {
+		// Proper use of the Kubernetes API Watch capability when no resourceVersion is supplied is to do a list first
+		// to get the initial state and a useable resourceVersion.
+		// See https://kubernetes.io/docs/reference/using-api/api-concepts/#efficient-detection-of-changes for details.
+		resourceVersion, err = kr.doInitialList(ctx, config, resource)
+		if err != nil {
+			kr.setting.Logger.Error("could not perform initial list for watch", zap.String("resource", config.gvr.String()), zap.Error(err))
+			return
+		}
+		// If we still don't have a resourceVersion we can try 1 as a last ditch effort.
+		// This also helps our unit tests since the fake client can't handle returning resource versions
+		// as part of a list of objects.
+		if resourceVersion == "" {
+			resourceVersion = defaultResourceVersion
+		}
+	}
+
 	watchFunc := func(options metav1.ListOptions) (apiWatch.Interface, error) {
-		return resource.Watch(ctx, metav1.ListOptions{
-			FieldSelector: config.FieldSelector,
-			LabelSelector: config.LabelSelector,
-		})
+		options.FieldSelector = config.FieldSelector
+		options.LabelSelector = config.LabelSelector
+		return resource.Watch(ctx, options)
 	}
 
-	watch, err := watch.NewRetryWatcher(config.ResourceVersion, &cache.ListWatch{WatchFunc: watchFunc})
+	watcher, err := watch.NewRetryWatcher(resourceVersion, &cache.ListWatch{WatchFunc: watchFunc})
 	if err != nil {
 		kr.setting.Logger.Error("error in watching object", zap.String("resource", config.gvr.String()), zap.Error(err))
 		return
 	}
 
-	res := watch.ResultChan()
+	res := watcher.ResultChan()
 	for {
 		select {
 		case data, ok := <-res:
 			if !ok {
 				kr.setting.Logger.Warn("Watch channel closed unexpectedly", zap.String("resource", config.gvr.String()))
 				return
 			}
-			logs := watchObjectsToLogData(&data, time.Now(), config)
-
-			obsCtx := kr.obsrecv.StartLogsOp(ctx)
-			err := kr.consumer.ConsumeLogs(obsCtx, logs)
-			kr.obsrecv.EndLogsOp(obsCtx, metadata.Type, 1, err)
+			logs, err := watchObjectsToLogData(&data, time.Now(), config)
+			if err != nil {
```

Comment:
Since we're passing around resourceVersion values properly there is now a chance that the client returns a 410 and an status object complaining that the resource doesn't exist. This is a rare situation that would normally only happen if the user self-supplied a resourceVersion that was too old.

At the moment I am handling that as an error and allowing the processor to keep running, but the group that had the error will stop. The proper way to handle that situation is to do another List to get a proper resourceVersion again.  I think we should do that, but as a followup PR.